californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -460,13 +380,6 @@ public void completed(final Exchange exchange) {
 					LOGGER.log(Level.FINER, "Exchange [{0}, {1}] completed", new Object[]{midKey, exchange.getOrigin()});
```

Comment:
if (!exchange.getCurrentRequest().getOptions().hasObserve()) {
					exchangeStore.releaseToken(idByToken);
				}

Same as in TcpMatcher, request with observe option 1 may be cleaned up on completion. rebase and use new Request.isObserve() ?